cram/cram_codecs.c

```diff
@@ -1416,25 +1410,36 @@ cram_codec *cram_byte_array_len_decode_init(char *data, int size,
     c->decode = cram_byte_array_len_decode;
     c->free   = cram_byte_array_len_decode_free;
     
-    cp += itf8_get(cp, &encoding);
-    cp += itf8_get(cp, &sub_size);
+    cp += safe_itf8_get(cp, endp, &encoding);
+    cp += safe_itf8_get(cp, endp, &sub_size);
+    if (sub_size < 0 || endp - cp < sub_size)
+        goto malformed;
     c->byte_array_len.len_codec = cram_decoder_init(encoding, cp, sub_size,
 						    E_INT, version);
+    if (c->byte_array_len.len_codec == NULL)
+        goto no_codec;
     cp += sub_size;
 
-    cp += itf8_get(cp, &encoding);
-    cp += itf8_get(cp, &sub_size);
+    cp += safe_itf8_get(cp, endp, &encoding);
+    cp += safe_itf8_get(cp, endp, &sub_size);
+    if (sub_size < 0 || endp - cp < sub_size)
```

Comment:
Similarly here there is no reset for the default values of encoding and sub_size, so they may just come from the ones above in the case of running out of 'cp'.